resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -373,28 +410,57 @@ private static void setSuccess(Promise<InetAddress> promise, InetAddress result)
         }
     }
 
+    boolean searchDomains(String hostname) {
+        if (searchDomains.length == 0 || (hostname.length() > 0 && hostname.charAt(hostname.length() - 1) == '.')) {
+            return false;
+        }
+        int idx = hostname.length();
+        int dots = 0;
+        while (idx-- > 0) {
+            if (hostname.charAt(idx) == '.') {
+                dots++;
+            }
+            if (dots >= ndots) {
```

Comment:
refactor this code so that we only check this condition when `dots` changes

```java
if (hostname.charAt(idx) == '.') {
  if (++dots >= ndots) {
    return false;
  }
}
```